surl/__init__.py

```diff
@@ -470,72 +440,21 @@ def main():
             url = "{}/v1/tokens/whoami".format(
                 CONSTANTS[config.store_env]["pubgw_base_url"]
             )
-        method = "GET"
-        data = None
-    else:
-        if args.data is not None:
-            if args.data.startswith("@"):
-                with open(os.path.expanduser(args.data[1:])) as fd:
-                    data = json.load(fd)
-            else:
-                data = json.loads(args.data)
-            method = args.method
-            if args.method == "GET":
-                method = "POST"
-        else:
-            data = None
-            method = args.method
 
     auth_header = get_authorization_header(config.root, config.discharge)
     headers.update(auth_header)
-    for h in args.headers:
-        try:
-            k, v = [t.strip() for t in h.split(":")]
-        except ValueError:
-            print('Invalid header: "{}"'.format(h))
-            return 1
-        headers[k] = v
-
-    if args.debug:
-        print(
-            "\033[1m******** request headers ********\033[0m",
-            file=sys.stderr,
-            flush=True,
-        )
-        for k, v in headers.items():
-            print("{}: {}".format(k, v), file=sys.stderr, flush=True)
-        print(
-            "\033[1m**********************************\033[0m",
-            file=sys.stderr,
-            flush=True,
-        )
 
-    response = store_request(
-        config, url=url, method=method, json=data, headers=headers, stream=True
-    )
+    # -s hides progress bar and errors, -S brings the errors back
+    arguments = ["curl", "-sS"]
 
-    if args.debug:
-        print(
-            "\033[1m******** response headers ********\033[0m",
-            file=sys.stderr,
-            flush=True,
-        )
-        print(
-            "HTTP/1.1 {} {}".format(response.status_code, response.reason),
-            file=sys.stderr,
-            flush=True,
-        )
-        for k, v in response.headers.items():
-            print("{}: {}".format(k, v), file=sys.stderr, flush=True)
-        print(
-            "\033[1m**********************************\033[0m",
-            file=sys.stderr,
-            flush=True,
-        )
+    for header, value in headers.items():
+        arguments.append("-H")
+        arguments.append(f"{header}: {value}")
+
+    arguments.extend(remainder)
+    arguments.append(url)
 
-    for chunk in response.iter_content(chunk_size=1024 * 8):
-        if chunk:
-            sys.stdout.buffer.write(chunk)
+    result = subprocess.run(arguments, stderr=subprocess.STDOUT)
 
     # Flush STDOUT carefully, because PIPE might be broken.
```

Comment:
Hi, I'm fairly certain this _noop method and the try/except below to flush stdout are not needed if we're letting curl handle all output; this seems to have been needed because the previous implementation did streaming read of the response and corresponding write by chunks to sys.stdout (see above diff lines 536-538 which were removed).

If you want to confirm, try downloading a snap, the issue that the above code fixed was https://github.com/canonical/surl/issues/8.